src/utils/common-utils.ts

```
@@ -232,3 +232,27 @@ export function calcAccuracyMultiplier(userAccStage: number, targetEvaStage: num
 
   return 1;
 }
+
+/**
+ * Returns the name of the key that matches the enum value.
+ *
+ * **Note**: **DO NOT** use this with TypeScript enums, only with `const` objects being used as enums.
+ * @param input - The enum to check
+ * @param val - The value to get the key of
+ * @returns The name of the key with the specified value
+ * @example
+ * const thing = {
+ *   one: 1,
+ *   two: 2,
+ * } as const;
+ * console.log(enumValueToKey(thing, thing.two)); // output: "two"
+ */
+export function enumValueToKey(input: object, val: number): string {
+  for (const [key, value] of Object.entries(input)) {
+    if (val === value) {
+      return key;
+    }
+  }
+  console.error("Invalid value passed to `enumValueToKey`!");
+  return "Invalid enum value!";
```

This should report an Error I'd say. Not some odd string return with a console-error only.
If you throw an error make sure to add `never` as return type

---

I wasn't sure if I should throw an error or not for this. I can switch it over.

---

Turns out that revealed a bug lmao, guess it's the right approach.